### PR TITLE
[CloudFront] Check specifically for undefined and determining of the defaultRootObject prop is set or not

### LINKED ISSUE #794: [CloudFront] Cannot leave "defaultRootObject" as an empty string
Wanting to set defaultRootObject to an empty string and CDK replaces it with "index.html". 

The check for this is `"props.defaultRootObject || "index.html",` which catches empty strings unintentionally 

```
new CloudFrontWebDistribution(this, 'a_name', {
      .................
      httpVersion: HttpVersion.HTTP2,
      priceClass: PriceClass.PriceClassAll,
      defaultRootObject: '',
    });
  }
```

### PR DESCRIPTION
**Pull request by ZeldoKavira**
Verify the variable is undefined and not an empty string since that is a valid setting.

Fixes #794 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 ## [UNRELEASED]
+
+
+### Bug Fixes
+
+* **aws-cloudfront:** Fix empty string defaultRootObject being replaced with the default value instead of passing the empty string to CloudFormation. Closes [#794](https://github.com/awslabs/aws-cdk/issues/794)
```
Comment (eladb) [inline CHANGELOG.md]: Not required. Just follow conventional commits and it will be added to the changelog

**Comment by ZeldoKavira** [comment]
Note: I cannot assign reviewers per the CONTRIBUTING documentation. (@eladb maybe?)

File: CHANGELOG.md (hunk 1 above)
Comment (ZeldoKavira) [reply CHANGELOG.md]: https://github.com/awslabs/aws-cdk/blob/master/CONTRIBUTING.md says it is required :(

**Comment by ZeldoKavira** [comment]
@eladb Added test (previous tests already cover default checking). Reverted changelog.

**Pull request merged**